modules/platform/nuxeo-platform-web-common/src/main/java/org/nuxeo/ecm/platform/web/common/idempotency/NuxeoIdempotentFilter.java

```diff
@@ -149,13 +149,15 @@ protected boolean doFilterIdempotent(HttpServletRequest request, HttpServletResp
                                 key, wrapper.getStatus(), content.length, MAX_CONTENT_SIZE);
                         return true;
                     }
+                    response.setHeader(HEADER_KEY, key);
```

Comment:
I did not find any better way to handle it given the fact that response is wrapped when being processed (for response copy) and because of the pattern to return a boolean, as asked by Florent, to avoid repeating calls to `chain.doFilter` whenever the request is not handled.
Note I did set this header either when the response is too big to be stored in the KV store (maybe it should still be added ?)

Reply:
OK.
> maybe it should still be added ?

I'd say yes